core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -786,20 +787,26 @@ class WorkspaceService(protected val userInfo: UserInfo,
             } else {
               None
             }
+            Option(WorkspaceListResponse(accessLevel, workspaceDetails, submissionStats, workspacePolicy.public))
 
-            WorkspaceListResponse(accessLevel, workspaceDetails, submissionStats, workspacePolicy.public)
+          } catch {
+            case ex: ApiException if ex.getCode == StatusCodes.NotFound.intValue => {
```

Comment:
Can you add a comment about why we are doing this? 

Also, we only intend to catch the exception in the case where the workspace is a MCWorkspace, so can you check that as well at line 793?

Reply:
Something along the lines of

```
case ex: ApiException if (WorkspaceType.McWorkspace == workspace.workspaceType) && (ex.getCode == StatusCodes.NotFound.intValue) => {

Reply:
To avoid multiple calls to `getWorkspace`, we use `list.mapFilter()` which returns an option and filters out all None values. 
The try catch block catches any ApiExceptions (specifically 404 errors) and returns None. `list.mapFilter()` will then filter out those workspaces and only return WorkspaceListResponse for workspaces that do not throw those errors.

Reply:
Sorry, I meant a comment in the code for future people who might be confused about why we are handling the 404 error in this way. Something like:

```
          // Internal folks may create MCWorkspaces in local WorkspaceManager instances, and those will not
          // be reachable when running against the dev environment.

Reply:
Got it, sorry I'll add that in the code!